cli/internal/run/run.go

```diff
@@ -1010,8 +1010,15 @@ func (e *execContext) exec(pt *packageTask, deps dag.Set) error {
 		outputs := pt.HashableOutputs()
 		targetLogger.Debug("caching output", "outputs", outputs)
 		ignore := []string{}
-		filesToBeCached := globby.GlobFiles(pt.pkg.Dir, outputs, ignore)
-		if err := e.turboCache.Put(pt.pkg.Dir, hash, int(time.Since(cmdTime).Milliseconds()), filesToBeCached); err != nil {
+		filesToBeCached := globby.GlobFiles(filepath.Join(e.rs.Opts.cwd, pt.pkg.Dir), outputs, ignore)
+		var relativePaths = make([]string, len(filesToBeCached))
+
+		for index, value := range filesToBeCached {
+			relativePath, _ := filepath.Rel(e.rs.Opts.cwd, value)
+			relativePaths[index] = relativePath
+		}
+
+		if err := e.turboCache.Put(pt.pkg.Dir, hash, int(time.Since(cmdTime).Milliseconds()), relativePaths); err != nil {
```

Comment:
The cache setup expects paths to be relative. To avoid needing to refactor the caching layer at this time, this resets the inputs to the previously expected behavior. It *should* be refactored to hide this as a cache implementation detail.

Reply:
My dream is that whatever the expectation ends up being is enforced by the type system.

Reply:
`AbsolutePath`, `ProjectRelativePath`, `WorkspaceRelativePath`, `CacheRelativePath`

🎉🎉🎉